src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -2120,12 +2120,17 @@ int bus_ensure_running(sd_bus *bus) {
 
         assert(bus);
 
-        if (IN_SET(bus->state, BUS_UNSET, BUS_CLOSED, BUS_CLOSING))
-                return -ENOTCONN;
         if (bus->state == BUS_RUNNING)
                 return 1;
 
+        /* sd_bus_process() may call user callbacks which may unref the bus.
+         * Make sure we still have a valid ref. */
+        BUS_DONT_DESTROY(bus);
```

Comment:
i think this should be dropped. The caller should make sure it has a ref on the bus for the whole time it calls this function. If it doesn't it's a bug in that, and it needs to add a ref. This function is basically too "far inside" already, and thus mostly would tape over bugs where the caller already fucked up, if you follow what I mean.